internal/plugins/helm/v1/scaffolds/api.go

```diff
@@ -70,26 +70,44 @@ func (s *apiScaffolder) scaffold() error {
 	if err != nil {
 		return err
 	}
-	r, chrt, err := chartutil.CreateChart(projectDir, s.opts)
+	resourceOptions, chrt, err := chartutil.CreateChart(projectDir, s.opts)
 	if err != nil {
 		return err
 	}
 
+	//todo(camilamacedo86): replace the options by kubernetes-sigs/kubebuilder#1974
+	if err := resourceOptions.Validate(); err != nil {
+		return err
+	}
+
 	// Check that resource doesn't exist
-	if s.config.HasResource(r.GVK()) {
+	if s.config.HasResource(resourceOptions.GVK()) {
 		return errors.New("the API resource already exists")
 	}
+
 	// Check that the provided group can be added to the project
-	if !s.config.MultiGroup && len(s.config.Resources) != 0 && !s.config.HasGroup(r.Group) {
+	if !s.config.IsMultiGroup() && s.config.ResourcesLength() != 0 && !s.config.HasGroup(resourceOptions.Group) {
 		return errors.New("multiple groups are not allowed by default, to enable multi-group set 'multigroup: true' in your PROJECT file")
 	}
 
-	res := r.NewResource(s.config, true)
-	s.config.UpdateResources(res.GVK())
+	resource := resourceOptions.NewResource(s.config)
+
+	resource.Domain = s.config.GetDomain()
```

Comment:
@Adirio is there a reason this isn't being set by `NewResource()`?

Reply:
https://github.com/kubernetes-sigs/kubebuilder/blob/72f01e58c677219d7af38385a1b249b496391167/pkg/plugins/golang/options.go#L126-L139

It sets the `Domain` in the `GVK` of the `Options`. Previously, when configs are injected to subcommands, `options.Domain = c.GetDomain()` is called. This is, the easliest we have the config we set it in the `Options` and that gets transfered to the `Resource`.

Reply:
Ah then this is redundant. @camilamacedo86 the scaffold templates should use `Resource.GVK.Domain`.

Reply:
Just a small comment: `Resource.Domain` can also be accessed directly as the `GVK` is actually embedded (and inlined for json)